test/soake2erekt/main_test.go

```diff
@@ -0,0 +1,254 @@
+package soake2erekt
+
+import (
+	"context"
+	"flag"
+	"log"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"knative.dev/eventing/test/rekt/resources/channel_impl"
+	"knative.dev/pkg/system"
+	pkgTest "knative.dev/pkg/test"
+	"knative.dev/reconciler-test/pkg/environment"
+	"knative.dev/reconciler-test/pkg/feature"
+	"knative.dev/reconciler-test/pkg/k8s"
+	"knative.dev/reconciler-test/pkg/knative"
+)
+
+// var global environment.GlobalEnvironment
+
+type SoakFlagsStruct struct {
+	Duration     time.Duration
+	Copies       int
+	PollInterval time.Duration
+	PollDuration time.Duration
+}
+
+var Flags SoakFlagsStruct
+
+var global environment.GlobalEnvironment
+
+// TestMain is the first entry point for `go test`.
+func TestMain(m *testing.M) {
+	channel_impl.EnvCfg.ChannelGK = "KafkaChannel.messaging.knative.dev"
+	channel_impl.EnvCfg.ChannelV = "v1beta1"
+
+	flag.DurationVar(&Flags.Duration, "soak-duration", 1*time.Hour, "Soak test duration")
+	flag.IntVar(&Flags.Copies, "soak-copies", 1, "Number of copies for each soak test scenario")
+	flag.DurationVar(&Flags.PollInterval, "soak-poll-interval", 5*time.Second, "Poll interval used in soak tests")
+	flag.DurationVar(&Flags.PollDuration, "soak-poll-duration", 10*time.Minute, "Poll duration used in soak tests")
+
+	restConfig, err := pkgTest.Flags.ClientConfig.GetRESTConfig()
+	if err != nil {
+		log.Fatal("Error building client config: ", err)
+	}
+
+	// Getting the rest config explicitly and passing it further will prevent re-initializing the flagset
+	// in NewStandardGlobalEnvironment().
+	global = environment.NewStandardGlobalEnvironment(func(cfg environment.Configuration) environment.Configuration {
+		cfg.Config = restConfig
+		return cfg
+	})
+
+	// Run the tests.
+	os.Exit(m.Run())
+}
+
+func soakTestEnvironment(t *testing.T, namespace string) (context.Context, environment.Environment) {
+	return global.Environment(
+		environment.WithNamespace(namespace),
+		knative.WithKnativeNamespace(system.Namespace()),
+		knative.WithLoggingConfig,
+		knative.WithTracingConfig,
+		k8s.WithEventListener,
+		// Enables KnativeService in the scenario.
+		//eventshub.WithKnativeServiceForwarder,
+		environment.WithPollTimings(Flags.PollInterval, Flags.PollDuration),
+		environment.WithTestLogger(t),
```

Comment:
Maybe I would add a comment to warn about adding "environment.Managed()" which would break all the stuff.